bindings/core/src/method_handler/wallet.rs

```diff
@@ -204,8 +205,20 @@ pub(crate) async fn call_wallet_method_internal(wallet: &Wallet, method: WalletM
             let implicit_account_creation_address = wallet.implicit_account_creation_address().await?;
             Response::Bech32Address(implicit_account_creation_address)
         }
-        WalletMethod::PrepareImplicitAccountTransition { output_id } => {
-            let data = wallet.prepare_implicit_account_transition(&output_id).await?;
+        WalletMethod::PrepareImplicitAccountTransition {
+            output_id,
+            public_key,
+            bip_path,
+        } => {
+            let data = if let Some(public_key_str) = public_key {
+                let public_key = PublicKey::try_from_bytes(prefix_hex::decode(public_key_str)?)
+                    .map_err(iota_sdk::wallet::Error::other)?;
```

Comment:
Why is this needed? Wallet has a From impl for crypto errors.

Reply:
Ah wait PublicKey is our type

Reply:
Hmm I think I would add a Types variant in Wallet + a from tbh, seems like something useful

Reply:
Ahhh there is `Block` but no from for it, maybe just add it?

Reply:
I'm confused, you mean add an error variant?

Reply:
it didn't help much but I fixed it

Reply:
But for my understanding, why can't we use ? directly ?